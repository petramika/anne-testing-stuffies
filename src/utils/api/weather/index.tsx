import { getWeatherData } from "./getWeather";

export {
    getWeatherData
};
