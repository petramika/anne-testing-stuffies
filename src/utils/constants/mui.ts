export enum VariantTypography {
    h1 = 'h1',
    h2 = 'h2',
    body1 = 'body1'
}