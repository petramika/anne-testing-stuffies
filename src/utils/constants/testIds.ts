export const CUSTOM_ELEMENT = 'custom-element';