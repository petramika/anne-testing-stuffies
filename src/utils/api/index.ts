import { getWeatherData } from "./weather/getWeather";

export {
    getWeatherData
};
