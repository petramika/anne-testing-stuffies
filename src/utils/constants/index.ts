import { VariantTypography } from "./mui";

export {
    VariantTypography
}