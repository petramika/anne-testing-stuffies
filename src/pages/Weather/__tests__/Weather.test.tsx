import { render, waitFor } from "@testing-library/react"
import { Weather } from '..'
import { describe, it, expect, vi, afterEach } from "vitest"
import { screen } from '@testing-library/dom'
import { SEARCH_RESPONSE_SUCCESS } from "./constants";
import { getWeatherData } from "utils/api";
import _ from "lodash";

describe("Integration tests Weather app", () => {
    afterEach(() => {
        vi.resetAllMocks()
      })
      
    it("Renders the weather page success search response", async () => { 
        // ARRANGE
        vi.mocked(getWeatherData).mockResolvedValue(SEARCH_RESPONSE_SUCCESS);
        
        // ACT
        const result = render(<Weather />)

        // ASSERT
        await waitFor(() => {
            expect(screen.getByText(_.first(SEARCH_RESPONSE_SUCCESS)!.name)).toBeDefined()
        })

        expect(result).toMatchSnapshot()

    })
})
