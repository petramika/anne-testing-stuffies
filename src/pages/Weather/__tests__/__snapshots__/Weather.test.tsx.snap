// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Integration tests Weather app > Renders the main page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
      >
        Vite + React
      </h1>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        Anne's testing stuffies
      </h2>
      <div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Introduce the Zip Code to get the city!
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="outlined input"
            id="outlined input-label"
          >
            Zip Code
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined input"
              type="text"
              value="L1 8JQ"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Zip Code
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="sunny weather"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            data-testid="custom-element"
            height="140"
            src="/images/sunny-weather.png"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            >
              Data for CP L1 8JQ
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  test name
                </p>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  test country
                </p>
              </div>
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Vite + React
    </h1>
    <h2
      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
    >
      Anne's testing stuffies
    </h2>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Introduce the Zip Code to get the city!
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="outlined input"
          id="outlined input-label"
        >
          Zip Code
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined input"
            type="text"
            value="L1 8JQ"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Zip Code
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <img
          alt="sunny weather"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          data-testid="custom-element"
          height="140"
          src="/images/sunny-weather.png"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
          >
            Data for CP L1 8JQ
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-0"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                test name
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                test country
              </p>
            </div>
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
